File: app/src/main/java/com/example/seki/training_recycler_view/MainActivity.java
package com.example.seki.training_recycler_view;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity implements RecyclerListFragment.FragmentListener {

    TextView mTitle;
    TextView mDescription;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mTitle = (TextView)findViewById(R.id.selected_title);
        mDescription = (TextView)findViewById(R.id.selected_description);

        if (savedInstanceState == null) {
            FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
            Fragment recyclerListFragment = RecyclerListFragment.getInstance();
            transaction.replace(R.id.container, recyclerListFragment);
            transaction.commit();
        }
    }

    @Override
    public void onClickItem(Data data) {
        mTitle.setText(data.title.get());
        mDescription.setText(data.description.get());

        int clickCount = Integer.parseInt(data.clickCount.get(), 10);
        data.clickCount.set(String.valueOf(++clickCount));
    }
}

File: app/src/main/java/com/example/seki/trainingRecyclerView/MainActivity.kt
package com.example.seki.trainingRecyclerView

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), RecyclerListFragment.FragmentListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        if (savedInstanceState == null) {
            val transaction = supportFragmentManager.beginTransaction()
            val recyclerListFragment = RecyclerListFragment.getInstance()
            transaction.replace(R.id.container, recyclerListFragment)
            transaction.commit()
        }
    }

    override fun onClickItem(data: Data) {
        label_title.text = data.title.get()
        label_description.text = data.description.get()

        var clickCount: Int = Integer . parseInt (data.clickCount.get(), 10)
        data.clickCount.set((++clickCount).toString())
    }
}
